backend/controllers/filtersController.js

```diff
@@ -0,0 +1,77 @@
+// itemController.js
+import Item from '../models/ItemModel.js';
+
+// Function to filter items
+// export const  filterItems = async (req, res) => {
+//     try {
+//         const { category, location, timeOrder } = req.query;
+
+//         let filterCriteria = {};
+
+//         // Add category filter if provided
+//         if (category) {
+//             filterCriteria.category = category.toLowerCase();
+//         }
+
+//         // Add location filter if provided
+//         if (location) {
+//             filterCriteria.location = location.toLowerCase();
+//         }
+
+//         // Query the database with the filter criteria
+//         let query = Item.find(filterCriteria);
+
+//         // Sort by time update if provided
+//         if (timeOrder === 'newest') {
+//             query = query.sort({ updatedAt: -1 });
+//         } else if (timeOrder === 'oldest') {
+//             query = query.sort({ updatedAt: 1 });
+//         }
+
+//         // Execute the query
+//         const items = await query.exec();
+
+//         // Return the filtered items
+//         res.json(items);
+//     } catch (error) {
+//         res.status(500).json({ message: 'Error fetching items', error });
+//     }
+// };
+
+
+export const filterItems = async (req, res) => {
+    try {
+        const { category, location, timeOrder } = req.query;
+
+        let filterCriteria = {};
+
+        // Add category filter if provided (case-insensitive)
+        if (category) {
+            filterCriteria.category = { $regex: new RegExp(category, 'i') };
+        }
+
+        // Add location filter if provided (case-insensitive)
+        if (location) {
+            filterCriteria.location = { $regex: new RegExp(location, 'i') };
+        }
+
+        // Query the database with the filter criteria
+        let query = Item.find(filterCriteria);
+
+        // Sort by time update if provided
+        if (timeOrder === 'newest') {
+            query = query.sort({ updatedAt: -1 });
+        } else if (timeOrder === 'oldest') {
+            query = query.sort({ updatedAt: 1 });
+        }
+
+        // Execute the query
+        const items = await query.exec();
+
+        // Return the filtered items
+        res.json(items);
+    } catch (error) {
+        res.status(500).json({ message: 'Error fetching items', error });
+    }
+};
+ 
```

Comment:
Consider adding additional filters like price range or donor type to improve search capabilities

Reply:
I don't think price range will be necessary since it's al donations for free and no items have prices